File: C++/2-Medium/00129._求根节点到叶节点数字之和/00129._求根节点到叶节点数字之和.cpp
#include <iostream>
using namespace std;

// 定义二叉树节点结构
struct TreeNode {
    int val;
    TreeNode *left, *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

// 辅助函数：递归求根到叶子节点数字之和
int sumNumbersHelper(TreeNode* root, int currentSum) {
    if (!root) return 0; // 如果当前节点为空，返回0
    currentSum = currentSum * 10 + root->val; // 更新路径值
    if (!root->left && !root->right) { // 如果是叶子节点，返回当前累计的和
        return currentSum;
    }
    // 递归左右子树
    return sumNumbersHelper(root->left, currentSum) + sumNumbersHelper(root->right, currentSum);
}

// 主函数：计算从根节点到叶节点的数字之和
int sumNumbers(TreeNode* root) {
    return sumNumbersHelper(root, 0);
}

int main() {
    // 构建示例二叉树: [1,2,3]
    TreeNode *root = new TreeNode(1);
    root->left = new TreeNode(2);
    root->right = new TreeNode(3);
    
    // 调用函数并输出结果
    cout << "The sum of root-to-leaf numbers is: " << sumNumbers(root) << endl;
    
    // 清理分配的内存
    delete root->left;
    delete root->right;
    delete root;
    
    return 0;
}

File: Python/2-Medium/00129._求根节点到叶节点数字之和/00129._求根节点到叶节点数字之和.py
from typing import Optional

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def sumNumbers(root: Optional[TreeNode]) -> int:
    """
    计算从根节点到叶子节点的所有路径所代表的数字之和。
    
    Args:
    - root: Optional[TreeNode], 二叉树的根节点。
    
    Returns:
    - int, 所有路径代表的数字之和。
    
    示例:
    >>> root = TreeNode(1, TreeNode(2), TreeNode(3))
    >>> sumNumbers(root)
    25
    """
    def dfs(node: Optional[TreeNode], current_sum: int) -> int:
        if not node:
            return 0
        current_sum = current_sum * 10 + node.val
        if not node.left and not node.right:
            return current_sum
        return dfs(node.left, current_sum) + dfs(node.right, current_sum)
    
    return dfs(root, 0)

# 构建示例二叉树
#     1
#    / \
#   2   3
root = TreeNode(1, TreeNode(2), TreeNode(3))

# 调用函数并打印结果
print(sumNumbers(root))  # 预期输出：25